src/test/java/de/retest/recheck/cli/subcommands/CommitIT.java

```diff
@@ -58,6 +58,9 @@ public void commit_should_not_accept_the_report_because_there_are_no_differences
 
 	@Test
 	public void commit_should_accept_and_update_the_sut_state_file() throws Exception {
+		System.setProperty( ProjectConfiguration.RETEST_PROJECT_ROOT, temp.getRoot().toString() );
```

Comment:
System properties in tests should be set via a mechanism that restores the original values after the tests have been executed, otherwise we may face weird side effects. System Rules has [`ProvideSystemProperty`](https://stefanbirkner.github.io/system-rules/#ProvideSystemProperty) for this.

Reply:
I would like to use the RestoreProperty System Rule. But this is currently not possible due to the implementation in recheck. There is already a branch that should fix the problem in recheck.